File: offer-cache-service/src/main/scala/pl/jugademy/offercache/infrastructure/KafkaConsumerFactory.scala
package pl.jugademy.offercache.infrastructure

import java.util.Properties

import org.apache.kafka.clients.consumer.KafkaConsumer
import org.apache.kafka.common.serialization.ByteArrayDeserializer

object KafkaConsumerFactory {
  def provide(config: Config): KafkaConsumer[Array[Byte], Array[Byte]] = {
    val props = new Properties()
    props.put("bootstrap.servers", config.getProperty("bootstrapServers"))
    props.put("group.id", config.getProperty("consumerGroup"))
    props.put("key.deserializer", classOf[ByteArrayDeserializer].getName)
    props.put("value.deserializer", classOf[ByteArrayDeserializer].getName)
    new KafkaConsumer(props)
  }
}

File: offer-search-service/src/main/java/pl/jugademy/offersearch/infrastructure/KafkaConsumerFactory.kt
package pl.jugademy.offersearch.infrastructure

import org.apache.kafka.clients.consumer.KafkaConsumer
import org.apache.kafka.common.serialization.ByteArrayDeserializer
import java.util.Properties

class KafkaConsumerFactory {
    companion object {
        fun provide(config: Config): KafkaConsumer<ByteArray, ByteArray> {
            val props = Properties()
            props["bootstrap.servers"] = config.getProperty<String>("bootstrapServers")
            props["group.id"] = config.getProperty<String>("consumerGroup")
            props["key.deserializer"] = ByteArrayDeserializer::class.java.name
            props["value.deserializer"] = ByteArrayDeserializer::class.java.name
            return KafkaConsumer(props)
        }
    }
}
